src/app.js

```diff
@@ -0,0 +1,139 @@
+import {div} from '@cycle/dom';
+import _ from 'lodash';
+import xs from 'xstream';
+
+// TO DO:
+// X Render Hello World
+// X Render board
+// X Render knight
+// X Move knight to legal squares by clicking
+// - Move knight to legal squares by dragging
+//
+
+function Board () {
+  return  _.range(0, 8).map(() =>
+    _.range(0, 8).map((square) => Square())
+  );
+}
+
+function Square () {
+  return {}
+}
+
+function isBlack (rowIndex, squareIndex) {
+  if (rowIndex % 2 === 0 && squareIndex % 2 !== 0) {
+    return true;
+  } else if (rowIndex % 2 !== 0 && squareIndex % 2 === 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function hasKnight (knightPosition, rowIndex, squareIndex) {
+  if (rowIndex === knightPosition.x && squareIndex === knightPosition.y) {
+    return true;
+  }
+
+  return false;
+}
+
+function legalMove (knightPosition, squareCoordinates) {
+  const moves = [
+    {x: -2, y: -1},
+    {x: -2, y: 1},
+    {x: 2, y: -1},
+    {x: 2, y: 1},
+    {x: -1, y: -2},
+    {x: -1, y: 2},
+    {x: 1, y: -2},
+    {x: 1, y: 2}
+  ]
+
+  const isLegal = moves.map(move => {
+    return _.isEqual(squareCoordinates, { x: move.x + knightPosition.x, y: move.y + knightPosition.y });
+  })
+  .filter(isLegal => isLegal === true)[0];
+
+  if (isLegal) { return true; }
+
+  return false;
+}
```

Comment:
This section is a little dense, and I think could benefit by being broken up a little.

Let's first begin by getting our potential moves:

```js
const potentialMoves = moves.map(move => ({ x: move.x + knightPosition.x, y: move.y + knightPosition.y }));
```

We can then check if it's legal using `Array.prototype.some`, which takes a selector function that returns a truthy result and returns `true` if any of the results are `true`, else `false`.

`[1, 2, 3].some(i => i % 2 == 0)` is the same as `[1, 2, 3].select(i => i % 2 === 0).length >= 1`

```js
const isLegal = potentialMoves.some(potentialMove => _.isEqual(potentialMove, squareCoordinates))
```

Then we can just:
```js
return isLegal
```

We could also just
```js
return potentialMoves.some(potentialMove => _.isEqual(potentialMove, squareCoordinates))
```

But sometimes it's nice to be explicit.